src/network/networkd-dhcp6.c

```diff
@@ -1058,11 +1110,13 @@ static int dhcp6_assign_delegated_prefix(Link *link,
         address->cinfo.ifa_prefered = lifetime_preferred;
         address->cinfo.ifa_valid = lifetime_valid;
 
-        link_set_state(link, LINK_STATE_CONFIGURING);
```

Comment:
dropped the setting state ?

Reply:
We never set link state like this in other places. Please see networkd-dhcp4.c or networkd-ndisc.c for example.

Reply:
This is the race we trying to fix. By defining different type of messages and a boolean value to set it true and check in ```link_check_ready```.

Reply:
After testing more, I noticed (but still I do not know why) a DHCPv6 address assigned in test-network.py (not systemd-networkd-tests.py) through dnsmasq stick to tentative state... So, setting `LINK_STATE_CONFIGURED` here or at one more place cause the link state stick to configuring state and never become to configured state.